kotest-assertions/kotest-assertions-shared/src/commonMain/kotlin/io/kotest/inspectors/Inspectors.kt

```diff
@@ -1,84 +1,173 @@
 package io.kotest.inspectors
 
-fun <T> Sequence<T>.forAll(fn: (T) -> Unit) = toList().forAll(fn)
-fun <T> Array<T>.forAll(fn: (T) -> Unit) = asList().forAll(fn)
-fun <T> Collection<T>.forAll(fn: (T) -> Unit) {
+fun <T> Sequence<T>.forAll(fn: (T) -> Unit): Sequence<T> {
+   toList().forAll(fn)
+   return this
+}
+
+fun <T> Array<T>.forAll(fn: (T) -> Unit): Array<T> {
+   asList().forAll(fn)
+   return this
+}
+
+fun <T, C : Collection<T>> C.forAll(fn: (T) -> Unit): C {
    val results = runTests(this, fn)
    val passed = results.filterIsInstance<ElementPass<T>>()
    if (passed.size < this.size) {
       val msg = "${passed.size} elements passed but expected ${this.size}"
       buildAssertionError(msg, results)
    }
+   return this
+}
+
+fun <T> Sequence<T>.forOne(fn: (T) -> Unit): Sequence<T> {
+   toList().forOne(fn)
+   return this
 }
 
-fun <T> Sequence<T>.forOne(fn: (T) -> Unit) = toList().forOne(fn)
-fun <T> Array<T>.forOne(fn: (T) -> Unit) = asList().forOne(fn)
-fun <T> Collection<T>.forOne(fn: (T) -> Unit) = forExactly(1, fn)
+fun <T> Array<T>.forOne(fn: (T) -> Unit): Array<T> {
+   asList().forOne(fn)
+   return this
+}
+
+fun <T, C : Collection<T>> C.forOne(fn: (T) -> Unit): C = forExactly(1, fn)
+
+fun <T> Sequence<T>.forExactly(k: Int, fn: (T) -> Unit): Sequence<T> {
+   toList().forExactly(k, fn)
+   return this
+}
+
+fun <T> Array<T>.forExactly(k: Int, fn: (T) -> Unit): Array<T> {
+   toList().forExactly(k, fn)
+   return this
+}
 
-fun <T> Sequence<T>.forExactly(k: Int, fn: (T) -> Unit) = toList().forExactly(k, fn)
-fun <T> Array<T>.forExactly(k: Int, fn: (T) -> Unit) = toList().forExactly(k, fn)
-fun <T> Collection<T>.forExactly(k: Int, fn: (T) -> Unit) {
+fun <T, C : Collection<T>> C.forExactly(k: Int, fn: (T) -> Unit): C {
    val results = runTests(this, fn)
    val passed = results.filterIsInstance<ElementPass<T>>()
    if (passed.size != k) {
-    val msg = "${passed.size} elements passed but expected $k"
-    buildAssertionError(msg, results)
-  }
+      val msg = "${passed.size} elements passed but expected $k"
+      buildAssertionError(msg, results)
+   }
+   return this
+}
+
+fun <T> Sequence<T>.forSome(fn: (T) -> Unit): Sequence<T> {
+   toList().forSome(fn)
+   return this
```

Comment:
FYI for more concise style you might think about something like this instead:
```
fun <T> Sequence<T>.forSome(fn: (T) -> Unit) = apply { toList().forSome(fn) }
```

Reply:
Yes I like that.

Reply:
Done